src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let stateClone = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        stateClone = {
+          ...stateClone,
+          ...extraData,
+        };
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of keysToRemove) {
+          delete stateClone[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        stateClone = {};
```

Comment:
You need to figure out how to clear the object differently, since now it is written as a `const`.
Create a function that will clear the object.
Move the function outside `transformStateWithClones` function.